### ISSUE TITLE
libcaf_opencl does not compile on windows

### ISSUE THREAD

**Issue by landagen** [actor-framework/actor-framework#644]
In trying to compile libcaf_opencl on my windows machine, I received a few errors.  

1. The pfn_notify function uses the __cdecl calling convention when the windows clCreateContext expects the __stdCall calling convention.

2. libcaf_opencl/test/opencl.cpp needs a cast to convert problem_size from a size_t to an int in two places

3.  the CL/*.h include files could not be found.  Adding "include_directories(AFTER ${OpenCL_INCLUDE_DIR})" when OpenCL_FOUND resolves the issue.

**Comment by landagen** [comment]
I have made the changes in my fork and it compiles for me on linux and windows, but I cannot test it because I do not have a graphics card that can run the examples.  If desired, I can create a pull request for it.

**Comment by Hamdor** [comment]
Could you give us some more informations about your system? I would like to know which windows version and graphics card (+driver version) has been used.

**Comment by landagen** [comment]
OS: Windows 10 Pro
Nvidia drivers version 378.66 (release date 12/20/2017)
GeForce GT 630M

Originally I could not compile against AMD, Nvidia, or intel openCL.  After making the changes I could compile against Nvidia and intel.  I haven't tested AMD.  

When I run the proper_matrix example, I get a runtime error of "clCreateContext: CL_DEVICE_NOT_AVAILABLE" so my belief is that my device is too old.  Looking on the web my card only support openCL 1.1.  I think it is time for me to get a new computer.

**Comment by landagen** [comment]
Ok.  I will give that a shot.  I also downloaded the Nvidia CUDA development kit, installed everything, and then compiled and ran their example.  I got an "unknown error" (30) when trying to obtain a device even though when it queried the device it found one that would work so I am not sure what is going on yet.  I will keep working to see if I can get it functioning on my machine.

**Comment by landagen** [comment]
I got the snippet to work with a few changes.  Here is the output

Found 2 platform(s).
* Intel(R) OpenCL
> Found 1 GPU(s).
GPU devices on the platfom:
>>> 'Intel(R) HD Graphics 4000'
  The number of parallel compute cores on the OpenCL device: '16'
  Max size in bytes of the arguments that can be passed to a kernel: '1024'.
  Maximum number of work-items in a work-group executing a kernel using the data parallel execution model: '512'
  Maximum dimensions that specify the global and local work-item IDs used by the data parallel execution model: '3'
  Maximum number of work-items that can be specified in each dimension of the work-group to clEnqueueNDRangeKernel: '(512, 512, 512)'
  OpenCL C version string: 'OpenCL C 1.2 '
  Is CL_FALSE if the implementation does not have a compiler available to compile the program source. Is CL_TRUE if the compiler is available: '1'
  Is CL_TRUE if the OpenCL device is a little endian device and CL_FALSE otherwise: '1'
  Size of global memory cache in bytes: '2097152'
  Size of global device memory in bytes: '1468006400'
  Size of local memory arena in bytes: '65536'
  Maximum configured clock frequency of the device in MHz: '1250'
  Max number of arguments declared with the __constant qualifier in a kernel: '8'
  Max size in bytes of a constant buffer allocation: '65536'
  Max size of memory object allocation in bytes: '367001600'
  Describes the resolution of device timer (in nanoseconds): '80'
  Device has the following properties:
   CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE [ ]
   CL_QUEUE_PROFILING_ENABLE              [x]
> Found 1 CPU(s).
CPU devices on the platfom:
>>> '       Intel(R) Core(TM) i7-3520M CPU @ 2.90GHz'
  The number of parallel compute cores on the OpenCL device: '4'
  Max size in bytes of the arguments that can be passed to a kernel: '3840'.
  Maximum number of work-items in a work-group executing a kernel using the data parallel execution model: '1024'
  Maximum dimensions that specify the global and local work-item IDs used by the data parallel execution model: '3'
  Maximum number of work-items that can be specified in each dimension of the work-group to clEnqueueNDRangeKernel: '(1024, 1024, 1024)'
  OpenCL C version string: 'OpenCL C 1.2 '
  Is CL_FALSE if the implementation does not have a compiler available to compile the program source. Is CL_TRUE if the compiler is available: '1'
  Is CL_TRUE if the OpenCL device is a little endian device and CL_FALSE otherwise: '1'
  Size of global memory cache in bytes: '262144'
  Size of global device memory in bytes: '2147352576'
  Size of local memory arena in bytes: '32768'
  Maximum configured clock frequency of the device in MHz: '2900'
  Max number of arguments declared with the __constant qualifier in a kernel: '480'
  Max size in bytes of a constant buffer allocation: '131072'
  Max size of memory object allocation in bytes: '536838144'
  Describes the resolution of device timer (in nanoseconds): '353'
  Device has the following properties:
   CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE [x]
   CL_QUEUE_PROFILING_ENABLE              [x]
> Found 0 accelerator(s).
* NVIDIA CUDA
> Found 1 GPU(s).
GPU devices on the platfom:
>>> 'GeForce GT 630M'
  The number of parallel compute cores on the OpenCL device: '2'
  Max size in bytes of the arguments that can be passed to a kernel: '4352'.
  Maximum number of work-items in a work-group executing a kernel using the data parallel execution model: '1024'
  Maximum dimensions that specify the global and local work-item IDs used by the data parallel execution model: '3'
  Maximum number of work-items that can be specified in each dimension of the work-group to clEnqueueNDRangeKernel: '(1024, 1024, 64)'
  OpenCL C version string: 'OpenCL C 1.1 '
  Is CL_FALSE if the implementation does not have a compiler available to compile the program source. Is CL_TRUE if the compiler is available: '1'
  Is CL_TRUE if the OpenCL device is a little endian device and CL_FALSE otherwise: '1'
  Size of global memory cache in bytes: '32768'
  Size of global device memory in bytes: '2147483648'
  Size of local memory arena in bytes: '49152'
  Maximum configured clock frequency of the device in MHz: '1320'
  Max number of arguments declared with the __constant qualifier in a kernel: '9'
  Max size in bytes of a constant buffer allocation: '65536'
  Max size of memory object allocation in bytes: '536870912'
  Describes the resolution of device timer (in nanoseconds): '1000'
  Device has the following properties:
   CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE [x]
   CL_QUEUE_PROFILING_ENABLE              [x]
> Found 0 CPU(s).
> Found 0 accelerator(s).
